### PR TITLE
operations labels for math, vec math nodes Patch 2

### PR DESCRIPTION
**Pull request by og76**
operations labels for math, vec math

math: 
+ node op labels (also show int/float)  
+ added invert,reciproc 
vector math: 
+ node op labels (hehe, the workaround..)  
+ added project(), reflect()

### CODE REVIEW

File: nodes/number/float_math.py [hunk 1]
```diff
@@ -39,6 +41,9 @@ class FloatMathNode(bpy.types.Node, AnimationNode):
     bl_label = "Math"
 
     def operationChanged(self, context):
+        for op in operationItems:
```
Comment (JacquesLucke) [inline nodes/number/float_math.py]: please don't use 'op' here. instead write "for item in operationItems"
I know that I wrote 'op' in the execution code but that was to improve readability. Here that is not the case (imo)

File: nodes/number/float_math.py [hunk 2]
```diff
@@ -51,6 +56,9 @@ def outputIntegerChanged(self, context):
     outputInteger = BoolProperty(name = "Output Integer", default = False,
         update = outputIntegerChanged)
 
+    labelOperation = StringProperty(name = "labelOperation", default = "A * B", update = operationChanged)
+    labelOutputType = StringProperty(name = "labelOutputType", default = " (float)", update = outputIntegerChanged)
```
Comment (JacquesLucke) [inline nodes/number/float_math.py]: the name attribute should be good for the ui, it doesn't have to look like a variable.  "Operation Label" would be better

File: nodes/number/float_math.py [hunk 3]
```diff
@@ -58,6 +66,9 @@ def create(self):
 
     def draw(self, layout):
         layout.prop(self, "operation", text = "")
+        
+    def draw_label(self):
```
Comment (JacquesLucke) [inline nodes/number/float_math.py]: uhh you're right here. In fact I didn't know about this function. We can leave it as it is now but most propably I will add here an abstraction layer as well. 
Then we will have the functions: 'draw', 'drawAdvanced' and 'drawLabel'

File: nodes/number/float_math.py [hunk 4]
```diff
@@ -70,6 +81,8 @@ def edit(self):
             self.outputInteger = False
 
     def recreateOutputSocket(self):
+        self.labelOutputType = " (integer)" if self.outputInteger else " (float)" #just to be more clear
```
Comment (JacquesLucke) [inline nodes/number/float_math.py]: maybe the 'labelOutputType' variable should just hold "Integer" or "Float". The brackets and spaces should be added in the draw_label function

File: nodes/vector/vector_math.py [hunk 5]
```diff
@@ -19,12 +21,17 @@ class VectorMathNode(bpy.types.Node, AnimationNode):
     bl_label = "Vector Math"
 
     def operationChanged(self, context):
+        for op in operationItems:
+            if op[0] == self.operation:
```
Comment (JacquesLucke) [inline nodes/vector/vector_math.py]: same with 'op' here

File: nodes/vector/vector_math.py [hunk 6]
```diff
@@ -19,12 +21,17 @@ class VectorMathNode(bpy.types.Node, AnimationNode):
...
+                self.labelOperation = op[2][:11]
         self.inputs["B"].hide = self.operation in operationsWithFloat
         self.inputs["Scale"].hide = self.operation not in operationsWithFloat
         executionCodeChanged()
 
     operation = EnumProperty(name = "Operation", items = operationItems, default = "ADD", update = operationChanged)
 
+    labelOperation = StringProperty(name = "labelOperation", default = "A + B", update = operationChanged)
```
Comment (JacquesLucke) [inline nodes/vector/vector_math.py]: same as above

File: nodes/number/float_math.py (hunk 3 above)
Comment (og76) [inline nodes/number/float_math.py]: do we really need an abstraction layer for this? I mean, how much should we abuse labels?
Plus, the current BL way works very well with custom labels (that prevail..), extra things to bother about.
Just wondering if it's worth the effort ..

Comment (JacquesLucke) [inline nodes/number/float_math.py]: The effort to do this is very low. 
max. 5 lines
I can never know wether we will use this more in the future. Also then this function blends better into the others. I rarely use underscores in variable and function names

**Pull request merged**